src/conversations/Conversation.ts

```diff
@@ -47,25 +64,114 @@ export class ConversationV1 {
 
   messagesPaginated(
     opts?: ListMessagesPaginatedOptions
-  ): AsyncGenerator<MessageV1[]> {
-    return this.client.listConversationMessagesPaginated(this.peerAddress, opts)
+  ): AsyncGenerator<DecodedMessage[]> {
+    return this.client.listEnvelopesPaginated(
+      [this.topic],
+      this.decodeMessage.bind(this),
+      opts
+    )
   }
 
   /**
    * Returns a Stream of any new messages to/from the peerAddress
    */
-  streamMessages(): Promise<Stream<MessageV1>> {
-    return this.client.streamConversationMessages(this.peerAddress)
+  streamMessages(): Promise<Stream<DecodedMessage>> {
+    return Stream.create<DecodedMessage>(
+      this.client,
+      [this.topic],
+      this.decodeMessage.bind(this)
+    )
+  }
+
+  async decodeMessage({
+    message,
+    contentTopic,
+  }: messageApi.Envelope): Promise<DecodedMessage> {
+    const messageBytes = fetcher.b64Decode(message as unknown as string)
+    const decoded = await MessageV1.fromBytes(messageBytes)
+    const { id, senderAddress, recipientAddress, sent } = decoded
+
+    // Filter for topics
+    if (
+      !senderAddress ||
+      !recipientAddress ||
+      !contentTopic ||
+      buildDirectMessageTopic(senderAddress, recipientAddress) !== this.topic
+    ) {
+      throw new Error('Headers do not match intended recipient')
+    }
+    const decrypted = await decoded.decrypt(this.client.legacyKeys)
+    const { content, contentType, error } = decodeContent(
+      decrypted,
+      this.client
+    )
+
+    return new DecodedMessage({
```

Comment:
Why isn't this calling DecodeMessage.fromV1Message()?

Reply:
That's a good question. Mostly because I wrote `fromV1Message` later. Fixed